src/BlazorWebView/src/core/RootComponent.cs

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Components;
+using Microsoft.AspNetCore.Components.WebView;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Components.WebView.Maui
+{
+	/// <summary>
+	/// Describes a root component that can be added to a <see cref="BlazorWebView"/>.
+	/// </summary>
+	public class RootComponent
+	{
+		/// <summary>
+		/// Gets or sets the CSS selector string that specifies where in the document the component should be placed.
+		/// This must be unique among the root components within the <see cref="BlazorWebView"/>.
+		/// </summary>
+		public string? Selector { get; set; }
+
+		/// <summary>
+		/// Gets or sets the type of the root component. This type must implement <see cref="IComponent"/>.
+		/// </summary>
+		public Type? ComponentType { get; set; }
```

Comment:
Consider making selector and component type get only and provide a constructor? That way you avoid checking for `IsNullOrWhitespace` and so on.

Reply:
This is used in XAML so it needs get/set.